Clarify how to find changelog for the RC candidates of providers (#36512)
mistercrunch/Airflow: dev/README_RELEASE_PROVIDER_PACKAGES.md

The status of testing the providers by the community is kept here:
<TODO COPY LINK TO THE ISSUE CREATED>

> The issue is also the easiest way to see which code affecting PRs are included in the RC candidates.

Reviewer: Nit: It would be better to rephrase `which code affecting PRs are included`

Author: Rephrased.

> The issue is also the easiest way to see which code affecting PRs are included in the RC candidates.

Author: ```suggestion
The issue is also the easiest way to see important PRs included in the RC candidates.
```
